### PR TITLE
Show valid remaining time

### PR DESCRIPTION
**Pull request by RadekVana**
BFW-1645

### CODE REVIEW

File: src/gui/screen_printing.cpp [hunk 1]
```diff
@@ -261,12 +266,15 @@ void screen_printing_data_t::enable_tune_button() {
     btn_tune.ico.Invalidate();
 }
 
-void screen_printing_data_t::update_remaining_time(time_t rawtime, uint16_t print_speed) {
-    w_etime_value.color_text = rawtime != time_t(-1) ? GuiDefaults::COLOR_VALUE_VALID : GuiDefaults::COLOR_VALUE_INVALID;
-    if (rawtime != time_t(-1)) {
-        if (print_speed != 100)
+void screen_printing_data_t::update_remaining_time(uint32_t millis, uint16_t print_speed) {
+    bool is_time_valid = int(millis) > 0;
```
Comment (espr14) [inline src/gui/screen_printing.cpp:270]: Is there a reason to convert to int?

**Review by espr14 — CHANGES_REQUESTED** [review]

File: src/gui/screen_printing.cpp (hunk 1 above)
Comment (DRracer) [reply src/gui/screen_printing.cpp:270]: yes, the time can become negative, which results in displying incorrect 37d+ of print time

Comment (DRracer) [inline src/gui/screen_printing.cpp:270]: `bool is_time_valid = millis > 0U;` ... if there was a sign comparison warning (which I doubt)?

Comment (DRracer) [reply src/gui/screen_printing.cpp:270]: but probably the whole unsigned int is a problem - either compare millis > 2^31 or keep millis as signed int32_t

Comment (RadekVana) [reply src/gui/screen_printing.cpp]: fixed

Comment (RadekVana) [reply src/gui/screen_printing.cpp]: changed, must be shorter than 365 days

**Review by espr14 — APPROVED** [review]

**Review by DRracer — APPROVED** [review]

**Pull request merged**